app/src/main/kotlin/com/wire/android/ui/calling/incoming/IncomingCallViewModel.kt

```diff
@@ -115,35 +124,47 @@ class IncomingCallViewModel @Inject constructor(
         }
     }
 
-    fun acceptCall() {
-        callRinger.stop()
+    private fun showJoinCallAnywayDialog() {
+        incomingCallState = incomingCallState.copy(shouldShowJoinCallAnywayDialog = true)
+    }
+
+    fun dismissJoinCallAnywayDialog() {
+        incomingCallState = incomingCallState.copy(shouldShowJoinCallAnywayDialog = false)
+    }
+
+    fun acceptCallAnyway() {
         viewModelScope.launch {
-            var backStackNode = ACCEPT_CALL_DEFAULT_BACKSTACK_NODE
-            var delayTime = ACCEPT_CALL_DEFAULT_DELAY_TIME
-            // if there is already an active call, then end it to accept the new incoming call
             establishedCallConversationId?.let {
                 endCall(it)
-                backStackNode = BackStackMode.UPDATE_EXISTED
-                delayTime = DELAY_TIME_AFTER_ENDING_CALL
+                delay(DELAY_END_CALL)
             }
-            observeIncomingCallJob.cancel()
+            acceptCall()
+        }
+    }
 
-            acceptCall(conversationId = incomingCallConversationId)
-            // We need to add some delay, for the case of joining a call while there is an active one,
-            // to get all values returned by avs callbacks
-            delay(delayTime)
-            navigationManager.navigate(
-                command = NavigationCommand(
-                    destination = NavigationItem.OngoingCall.getRouteWithArgs(listOf(incomingCallConversationId)),
-                    backStackMode = backStackNode
+    fun acceptCall() {
+        viewModelScope.launch {
+            if (incomingCallState.hasEstablishedCall) {
+                showJoinCallAnywayDialog()
+            } else {
+                callRinger.stop()
+
+                dismissJoinCallAnywayDialog()
+                observeIncomingCallJob.cancel()
+
+                acceptCall(conversationId = incomingCallConversationId)
+
+                navigationManager.navigate(
+                    command = NavigationCommand(
+                        destination = NavigationItem.OngoingCall.getRouteWithArgs(listOf(incomingCallConversationId)),
+                        backStackMode = BackStackMode.REMOVE_CURRENT_AND_REPLACE
+                    )
                 )
-            )
+            }
         }
     }
 
     companion object {
-        private const val DELAY_TIME_AFTER_ENDING_CALL = 1000L
-        private val ACCEPT_CALL_DEFAULT_BACKSTACK_NODE = BackStackMode.REMOVE_CURRENT
-        private const val ACCEPT_CALL_DEFAULT_DELAY_TIME = 200L
+        const val DELAY_END_CALL = 200L
```

Comment:
Here we are using `200L` for a default delay when accepting a call where as the delay time for after ending a call before was `1000L`, should we keep `200L`?

Reply:
Yes it should be fine.. It's done differently now the delay is added just after ending the call